arch/arm/core/aarch32/exc_exit.S

```diff
@@ -73,11 +74,17 @@ SECTION_SUBSEC_FUNC(TEXT, _HandlerModeExit, z_arm_int_exit)
 SECTION_SUBSEC_FUNC(TEXT, _HandlerModeExit, z_arm_exc_exit)
 
 #ifdef CONFIG_PREEMPT_ENABLED
-    ldr r0, =_kernel
+    ldr r3, =_kernel
```

Comment:
although this seems ok - why did you have to change the register here?

Reply:
mainly so that I can free up `r0` and `r1` for use later (see below this line) while keeping the `_kernel` pointer in `r3`.

Reply:
couldn't you keep kernel in r0 and do 

```
  ldr r3, [r0, #_kernel_offset_to_nested]
  cmp r3, #1
```

Reply:
I prefer to keep the values that are referenced multiple times in the upper registers (in terms of caller-saved, that is) and use the lower registers as scratch.

That seems to be the convention used throughout the Zephyr ARM arch code as well (we are loading `_kernel` into `r2` and using `r0` and `r1` as scratch registers most of the time).

Reply:
never thought there's actually a convention about it, and I am certainly did not following it myself :)

OK, I was mostly worried that I am missing something here, but if this is a matter of preference, then i won't object